test/atom_finder/classifier_test.clj

```diff
@@ -70,74 +70,113 @@
+                
 
     (is (= false
            (->> "conditional.c" resource-path tu
                 (get-in-tree [0 2 1 0 1 1])
-                conditional-atom?
-                )))
+                conditional-atom?)))
+                
 
     (let [lines  (->> (resource-path "conditional.c")
                       tu
                       (atoms-in-tree conditional-atom?)
                       (map loc)
                       (map :line))]
 
-      (is (= lines [4 28 28 28 61]))
-    )))
+      (is (= lines [4 28 28 28 61])))))
+    
 
 (deftest preprocessors-in-context-test
   (testing "Macro entirely in context"
     (let [filename (resource-path "macro-in-expression.c")
           atoms (preprocessors-in-contexts all-preprocessor non-toplevel-classifier (tu filename))]
 
-      (is (= (map :line atoms) '(5 8 11)))
-      ))
+      (is (= (map :line atoms) '(5 8 11)))))
+      
 
   (testing "Macro starts in context"
     (let [filename (resource-path "if-starts-in-expression.c")
           atoms (preprocessors-in-contexts all-preprocessor non-toplevel-classifier (tu filename))]
 
-      (is (= (map :line atoms) '(9 11 14 16 18 23))) ; technically 25 should be here too because otherwise it's dependent on which if branch is evaluated
-      ))
+      (is (= (map :line atoms) '(9 11 14 16 18 23))))) ; technically 25 should be here too because otherwise it's dependent on which if branch is evaluated
+      
 
   (testing "Macro applied in function"
     (let [filename (resource-path "macro-application.c")
           atoms (preprocessors-in-contexts all-preprocessor non-toplevel-classifier (tu filename))]
 
-      (is (= (map :line atoms) '()))
-      ))
+      (is (= (map :line atoms) '()))))
+      
 
   (testing "Preprocessor from snippet study"
     (let [filename (resource-path "define-in-if-loop.c")
           atoms (preprocessors-in-contexts all-preprocessor statement-expression-classifier (tu filename))]
 
-      (is (= (map :line atoms) '(4 7 11)))
-      ))
+      (is (= (map :line atoms) '(4 7 11)))))
+      
 
   (testing "preprocesor-in-statement classifier"
     (let [pisc (-> atom-lookup (strict-get :preprocessor-in-statement) :classifier)]
       (is (true? (pisc (parse-expr "1 + \n#define M\n 3"))))
       (is (false? (pisc (parse-stmt "1 + \n#define M\n 3;"))))
-      (is (false? (pisc (parse-stmt "{ 1 + 3; f(); }"))))
-      )))
+      (is (false? (pisc (parse-stmt "{ 1 + 3; f(); }")))))))
+      
 
 (deftest literal-encoding-test
   (testing "utils"
     (->>
      [[:dec "1234"] [:oct "01234"] [:hex "0x1234"] [:bin "0b1234"]]
      (map (fn [[k s]]
             (is (= k (radix s)))
-            (is (= k (radix (parse-expr s))))
-            ))
-    )))
+            (is (= k (radix (parse-expr s)))))))))
+
+(deftest comma-operator-test
+  (testing "small statements" 
+    (is (comma-atom? (parse-expr "1,2")))
+    (is (false? (comma-atom? (parse-expr "int a,b;"))))
+    (is (comma-atom? (parse-expr "num1,num2")))))
```

Comment:
These tests are all fine, but you can do better. For example, you are trying to commit a file called comma.c, but you never use it in this test... could you?